### PR TITLE
Value Add/Waste Time functionality for analytics

### PR DESCRIPTION
**Pull request by dangond-ptc**
Apologies for the massive mono-commit! This brings in the ability to mark parts of the analytics timeline as being Value Add or Waste Time. Right now, the only visualization of this is as regions on the timeline, but I'm looking into where a good spot to put a summary would be. Depends on https://github.com/ptcrealitylab/vuforia-spatial-core-addon/pull/289 and https://github.com/ptcrealitylab/vuforia-spatial-edge-server/pull/924

### CODE REVIEW

**Comment by dangond-ptc** [comment]
Also, this has the unfortunate side-effect of not being backwards-compatible with existing analytics sessions. I'll also add in a commit that can change the previous data format into the new one.

File: src/utilities/Timeline.js [hunk 1]
```diff
@@ -0,0 +1,361 @@
+/**
+ * A Region represents a chunk of time [startTime, endTime) with a given value.
+ */
+export class Region {
```
Comment (benptc) [inline src/utilities/Timeline.js:4]: I like how nicely structured the Region and Timeline classes are – nice work!

File: src/utilities/Timeline.test.js [hunk 2]
```diff
@@ -0,0 +1,346 @@
+import { expect, test, describe } from 'vitest';
+import { Region, Timeline } from "./Timeline";
+
+const region = new Region(0, 5, null);
+
+describe('region', () => {
```
Comment (benptc) [inline src/utilities/Timeline.test.js:6]: Nice to see the tests here, too!

**Review by hobinjk-ptc — APPROVED** [review]
Very nice API design! Even with tests! Incredible

**Review by benptc — APPROVED** [review]
Looks good!

**Pull request merged**